### PR TITLE
merged-with-ben

### PR DESCRIPTION
**Pull request by bjones-brandcollective**

### CODE REVIEW

**Review by BenjaminTG — COMMENTED** [review]
Looks fine in terms of my limited exposure to React.

File: src/components/orderDetails.js [hunk 1]
```diff
@@ -10,12 +10,56 @@ function prettyDate(rawDate) {
...
+class RefundTotal extends React.Component {
+  constructor(props) {
+      super(props)
+  }
+  calRefundTotal(){
+    const qty = this.props.qty;
+    const price = 100;
+    const result = price * qty;
+    return result;
+  }
```
Comment (dkerr-brandcollective) [inline src/components/orderDetails.js:36]: Good for now but will need updating in future -- as it is currently set out, this function assumes that all items have the same price.

One approach you can take is to get rid of the `qty` prop, and replace it with an `returnDetails` prop (or something similar), where you plug in the return details information from the parent component. (E.g., `<RefundTotal return-details={this.state.returnDetails} />`).

Then, in this component, `calRefundTotal()` can look at each of the lines in the `returnDetails` prop, multiply the quantity by the price, and figure out the grand total. For example:

```js
calRefundTotal() {
  let grandTotal = 0;
  this.props.returnDetails.items.map(item => {
    grandTotal = grandTotal + (item.qty * item.price);
  })
  return grandTotal;
}
```

You don't need to update this for this commit -- this note is here for the future.

File: src/components/orderDetails.js [hunk 2]
```diff
@@ -10,12 +10,56 @@ function prettyDate(rawDate) {
...
+}
+
 class OrderDetails extends React.Component {
     
     constructor(props) {
         super(props)
+        this.state = {
+            quantityToReturn: 0,
+            quantityToExchange: 0,
```
Comment (dkerr-brandcollective) [inline src/components/orderDetails.js:52]: If the `<OrderDetails />` component contains all of the items within the order, it should probably contain a `returnDetails` object in its state. Storing the quantity to return and the quantity to exchange is going to make it hard to figure out how many of which specific item will be returned/exchanged.

The `returnDetails` object will probably end up looking like something similar to the return details object in [the documentation](https://docs.google.com/document/d/1xBk5xdV34KGzlhiBqPHqllHVhu7BIynAZUYEQXGgPyQ/edit#heading=h.7r5zg59skhcw), for example:

```json
{
    "return_items": [
        {
            "level_code": "ABC_1234",
            "quantity": 1,
            "reason": "FAULTY",
            "photos": [
                "75801619-794b-477c-b495-2f268dfbcbfd",
                "79bda7e6-d829-4dfd-ae58-4e12a2856eaa",
                "a810570f-7497-4b15-8fc0-95e20b77cede"
            ],
            "comments": "The sole is missing from the shoe"
        }
    ]
}
```

This comment is not for you to act on now (this merge can go through) -- this is for future reference.

File: src/components/orderItem.js [hunk 3]
```diff
@@ -18,7 +41,7 @@ class OrderItem extends React.Component {
                 <div className="uk-width-1-2">
                     <div className="uk-grid">
                         <div className="uk-width-1-3">
-                            <img alt={this.props.item.image} src={this.props.item.image} width="30em" height="30em" />
+                            <img src={this.props.item.image} width="30em" height="30em" />
```
Comment (dkerr-brandcollective) [inline src/components/orderItem.js:44]: Good catch on the alt tag.
Might want to add an alt tag using the product name, for example:

```jsx
<img alt={"Photo of " + this.props.item.name} src={this.props.item.image} ... />
```

File: src/components/orderItem.js [hunk 4]
```diff
@@ -31,27 +54,32 @@ class OrderItem extends React.Component {
...
+                                choices={elements}
+                                onChoice={(choice) => this.returnExchangeDetails(choice)} />
+                                <p>{this.props.item.validation_errors}</p>
+                            </div>
                               }
-                              {this.props.item.quantity_return_allowed === 0 &&
+                              {this.props.item.quantity_return_allowed == 0 &&
                                 <p>
-                                This item isn't returnable sorry
+                                    {!!(this.props.item.validation_errors) ? this.props.item.validation_errors : "" }
```
Comment (dkerr-brandcollective) [inline src/components/orderItem.js:68]: Good for now. Will eventually need to "translate" these validation errors.
Validation Errors is an array, too, so the items will need to be looped over to display correctly.

**Review by dkerr-brandcollective — APPROVED** [review]

**Pull request merged**